core/src/lib.rs

```diff
@@ -220,6 +254,35 @@ pub struct CoreRuntime {
     telemetry: TelemetryInstance,
     runtime: Option<tokio::runtime::Runtime>,
     runtime_handle: tokio::runtime::Handle,
+    shared_namespace_map: Arc<Mutex<HashMap<ClientIdentity, SharedNamespaceWorker>>>,
+    task_queue_key: Uuid,
+    heartbeat_interval: Option<Duration>,
+}
+
+/// Holds telemetry options, as well as worker heartbeat_interval.
+#[non_exhaustive]
+pub struct RuntimeOptions {
+    telemetry_options: TelemetryOptions,
+    heartbeat_interval: Option<Duration>,
+}
+
+impl RuntimeOptions {
+    /// Creates new runtime options.
```

Comment:
Since this is non_exhaustive it should probably use a builder pattern rather than a new